node/src/services/blockchain/transaction.rs

```diff
@@ -0,0 +1,150 @@
+use std::{
+    str::FromStr,
+    time::{Duration, Instant},
+};
+
+use anyhow::anyhow;
+use log::*;
+use sp_core::H256;
+use storage_hub_infra::actor::ActorHandle;
+use tokio::sync::mpsc::Receiver;
+
+use crate::services::blockchain::{commands::BlockchainServiceInterface, types::ExtrinsicResult};
+
+use super::{types::ExtrinsicHash, BlockchainService};
+
+const LOG_TARGET: &str = "blockchain-transaction";
+
+#[derive(Debug)]
+pub struct SubmittedTransaction {
+    watcher: Receiver<String>,
+    hash: ExtrinsicHash,
+    timeout: Option<Duration>,
+}
+
+const NO_TIMEOUT_WARN: Duration = Duration::from_secs(60);
```

Comment:
AFAIU 60s is arbitrary. `timeout()` necessarily expects some value and this constant is a default one, so that after it passes, in case there is no timeout set (which would yield error), we just warn "ok, I waited 60s, I'm gonna try again" without waiting indefinitely at that point.

Reply:
That is, we _will_ wait indefinitely if there is no timeout, but there will be warnings saying what's going on.

Reply:
@links234 maybe call the constant `NO_TIMEOUT_INTERVAL_WARNING`? If you don't think it's a mouthful 😆